README.md

```diff
@@ -1,33 +1,195 @@
-* Functional dependencies, both exact and approximate
-* Conditional functional dependencies
-* Association rules
+Desbordante can be used via three interfaces:
+* **Console application.** This is a classic command-line interface that aims to provide basic profiling functionality, i.e. discovery and validation of patterns. A user can specify pattern type, task type, algorithm, input file(s) and obtain results to the screen or into an output file.
+* **Python bindings.** Desbordante functionality can be accessed from within Python programs by employing the Desbordante Python library. This interface offers everything that is currently provided by the console version and allows advanced use, such as building interactive applications and designing scenarios for solving some particular real-life task. Relational data processing algorithms accept Pandas dataframes as input, allowing the user to conveniently preprocess data before mining patterns.
+* **Web interface.** There is a web interface that provides discovery and validation tasks with a rich interactive interface with visualization. However, currently it supports a limited number of patterns and should be considered more as an interactive demo.
 
-It also allows to run data cleaning scenarios using these algorithms. At the moment, we have  implemented a typo detection scenario using an exact and approximate functional dependency discovery algorithm.
+A brief introduction into the tool and its use cases is presented [here](https://medium.com/@chernishev/exploratory-data-analysis-with-desbordante-4b97299cce07) (in English) and [here](https://habr.com/ru/company/unidata/blog/667636/) (in Russian). Also, a list of various articles and guides can be found [here](https://desbordante.unidata-platform.ru/papers).
 
-The algorithms of Desbordante are implemented in C++ to maximize the resulting performance. They can be run using either a console version or a web-application that features an easy-to-use web interface.
+# Console
 
-You can try the deployed version [here](https://desbordante.unidata-platform.ru/). You have to register in order to process your own datasets. Keep in mind that due to a large demand various time and memory limits are enforced (and a task is killed if it goes outside of acceptable ranges).
+Usage examples:
+1) Discover all exact functional dependencies in a table represented by a .csv file that uses a comma as the separator and has a header row. In this example the default FD discovery algorithm (HyFD) is used.
 
-A brief introduction into the tool and its use-cases is presented [here](https://medium.com/@chernishev/exploratory-data-analysis-with-desbordante-4b97299cce07) (in English) and [here](https://habr.com/ru/company/unidata/blog/667636/) (in Russian).
+```sh
+--task=fd --table=../examples/datasets/university_fd.csv , True
```

Comment:
This is just listing the options, but the command the user should issue is going to be `python3 cli.py ...` or `./cli.py ...`, so that's what should be here (applies to the following console scenarios as well)